### PR TITLE
[BACKLOG-37991] WCAG - Manage Data souces dialog on Safari - bad ui format

### PR DESCRIPTION
**Pull request by varuntangirala**
@pentaho/wcag Please review and merge along with
https://github.com/pentaho/pentaho-platform/pull/5529

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 7m 4s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **345**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-1020/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by rmansoor — APPROVED** [review]

File: widgets/src/main/java/org/pentaho/gwt/widgets/client/table/BaseTable.java [hunk 1]
```diff
@@ -489,13 +501,26 @@ public void setFocusableRow( int row, boolean focus ) {
     }
   }
 
+  protected class MyScrollTable extends ScrollTable {
```
Comment (singletonc) [inline widgets/src/main/java/org/pentaho/gwt/widgets/client/table/BaseTable.java:504]: Not a blocker. Consider a more specific class name.

**Review by singletonc — APPROVED** [review]

File: widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css [hunk 2]
```diff
@@ -8535,6 +8535,50 @@ div.dijitDateTextBox.dijitTextBox .dijitReset.dijitRight.dijitButtonNode.dijitAr
...
+
+.safari #datasourceAdminDialog .gwt-ScrollTable .headerTable tr > td {
+  border-width: 0;
+}
+
+.safari #datasourceAdminDialog .gwt-ScrollTable .headerTable tr > td:last-of-type {
+  border-left-width: 1px;
+}
+
+.safari #datasourceAdminDialog .gwt-ScrollTable :is(.headerWrapper, .dataWrapper) {
```
Comment (singletonc) [inline widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css:8562]: Some of these rules are redundant and could be combined:
`selector1,
selector2,
selector3 {
styles
}`

File: widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css [hunk 3]
```diff
@@ -8535,6 +8535,50 @@ div.dijitDateTextBox.dijitTextBox .dijitReset.dijitRight.dijitButtonNode.dijitAr
...
+  box-sizing: border-box;
+}
+
+.safari #datasourceAdminDialog .gwt-ScrollTable :is(.headerWrapper .headerTable, .dataWrapper .dataTable) td:last-child {
+  padding-right: 10px;
+}
+/* end for styling manage data sources dialog for safari browser */
+
 :where(.responsive#general-datasource-window) #general-settings-box {
   box-sizing: border-box;
```
Comment (singletonc) [inline widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css:8583]: could add the border-box here or even create/look for a rule that adds the border-box style to any element that has that class

File: widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css (hunk 2 above)
Comment (dcleao) [reply widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css:8562]: Can you clarify which ones? Not spotting.

File: widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css (hunk 3 above)
Comment (dcleao) [reply widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css:8583]: We group CSS properties under the same CSS rule, using the "OR" combinator, within the same context / component. In this case, I think that keeping the safari rules in the "Safari override context" is more important.
I agree this is debatable 👍

**Pull request merged**